src/linker/Linker/MessageContainer.cs

```diff
@@ -67,6 +67,9 @@ public static MessageContainer CreateWarningMessage (LinkContext context, string
 			if (!(code > 2000 && code <= 6000))
 				throw new ArgumentException ($"The provided code '{code}' does not fall into the warning category, which is in the range of 2001 to 6000 (inclusive).");
 
+			if (context.WarningSuppressor != null && origin.MemberDefinition != null)
```

Comment:
We must only write suppressions for warnings which are actually reported - so this has to go after the `IsWarningSupressed` below.